python/lsst/ctrl/bps/workflow/HTCondor/htcondor_engine.py

```diff
@@ -100,24 +100,27 @@ def _handle_job_cmds(task_attrs, jobcmds):
 
         def _handle_job_inputs(gen_workflow, jobcmds):
             """Add job input files from generic workflow to job
+
             Parameters
             ----------
             gen_workflow : `networkx.DiGraph`
                 The generic workflow graph (e.g., has executable name and arguments)
             jobcmds : `RestrictedDict`
                 Commands for a htcondor job, updated with transfer of job inputs
             """
-            # inputs = [file_id for file_id in self.workflow_graph.predecessors(task_id)]
-            transinputs = []
-            # for file_id in inputs:
+            inputs = []
             for file_id in gen_workflow.predecessors(task_id):
                 file_attrs = gen_workflow.nodes[file_id]
                 is_ignored = file_attrs.get("ignore", False)
                 if not is_ignored:
-                    transinputs.append(file_attrs["pfn"])
-
-            if len(transinputs) > 0:
-                jobcmds["transfer_input_files"] = ",".join(transinputs)
+                    try:
+                        inputs.append(file_attrs["pfn"])
+                    except KeyError:
+                        _LOG.error("Missing pfn in %s-%s's file_attrs: %s", task_id, file_id, str(file_attrs))
+                        raise
```

Comment:
I do like this code more and would actually match my coding style more, but I also get told that checking first that a key exists isn't pythonic.  One should just try to access the key and catch exceptions.

Reply:
Just because exceptions are a thing doesn't mean we should always use them. You see a lot of people using `dict.get()` specifically because it doesn't raise. I have no problem with using `not in` checks if it results in cleaner looking code.